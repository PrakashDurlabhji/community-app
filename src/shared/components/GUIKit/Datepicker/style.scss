@import '~components/GUIKit/Assets/Styles/default';

.container {
  position: relative;
  display: flex;
  flex-direction: column;
  flex-grow: 1;

  .label {
    @include textInputLabel;
  }

  &.haveValue .label,
  &.isFocused .label {
    display: flex;
  }

  &.isFocused .label {
    color: $gui-kit-level-2;
  }

  &.haveError .label,
  &.haveError.isFocused .label {
    color: $gui-kit-level-5;
  }

  :global {
    @import '~react-dates/lib/css/_datepicker.css';

    .SingleDatePicker {
      width: 100%;
      position: relative;
      display: inline-block;
    }

    button {
      &:focus {
        outline: none;
      }
    }

    .SingleDatePickerInput {
      position: relative;
      border: none;
      width: 100%;

      .SingleDatePickerInput_calendarIcon {
        position: absolute;
        top: 12px;
        right: 0;
        margin: 0;
        padding: 0;
        width: 48px;
        bottom: 0;
        z-index: 1;
        background: 0 0;
        border: 0;
        color: inherit;
        font: inherit;
        line-height: normal;
        overflow: visible;
        cursor: pointer;
        display: inline-block;
        vertical-align: middle;
      }

      .DateInput {
        padding: 0;
        width: 100%;
        font-family: "Roboto", Helvetica, Arial, sans-serif;
        font-weight: 400;
        font-size: 15px;
        margin: 0;
        background: #fff;
        position: relative;
        display: inline-block;
        vertical-align: middle;

        input {
          @include textInput;

          height: 52px;
          padding-right: 48px;
        }

        .DateInput_fang {
          display: none;
        }

        .DateInput_screenReaderMessage {
          border: 0;
          clip: rect(0, 0, 0, 0);
          height: 1px;
          margin: -1px;
          overflow: hidden;
          padding: 0;
          position: absolute;
          width: 1px;
        }
      }

      .SingleDatePicker_picker {
        z-index: 7;
        top: 64px !important;
        background-color: #fff;
        position: absolute;

        .DayPicker {
          background: #fff;
          position: relative;
          text-align: left;

          .DayPicker_weekHeaders__horizontal {
            margin-left: 9px;

            .DayPicker_weekHeader {
              color: #757575;
              position: absolute;
              top: 62px;
              z-index: 2;
              text-align: left;

              .DayPicker_weekHeader_ul {
                list-style: none;
                margin: 1px 0;
                padding-left: 0;
                padding-right: 0;
                font-size: 14px;

                .DayPicker_weekHeader_ul {
                  list-style: none;
                  margin: 1px 0;
                  padding-left: 0;
                  padding-right: 0;
                  font-size: 14px;
                  display: inline-block;
                  text-align: center;
                }
              }
            }
          }

          .DayPicker_weekHeaders {
            position: relative;

            .DayPicker_weekHeader_li {
              font-size: 15px;
              font-family: "Roboto", Helvetica, Arial, sans-serif;
              font-weight: 400;
              line-height: 20px;
              display: inline-block;
              text-align: center;
            }
          }

          .DayPicker_focusRegion {
            outline: 0;
          }
        }

        .DayPicker__withBorder {
          border: 1px solid $gui-kit-gray-30;
          box-shadow: 2px 2px 3px 0 $tc-gray-neutral-light;
          overflow: hidden;
          border-radius: 3px;
        }

        .DayPickerNavigation__horizontal {
          height: 0;
        }

        .DayPickerNavigation {
          position: relative;
          z-index: 2;
        }

        .DayPickerNavigation_button {
          position: absolute;
          top: 18px;
          cursor: pointer;

          &:focus {
            outline: none;
          }

          &:first-child {
            left: 18px;
          }

          &:last-child {
            right: 18px;
          }
        }

        .CalendarMonth_caption {
          font-size: 16px;
          line-height: 26px;
          padding-top: 13px;
          padding-bottom: 10px;
          margin-bottom: 45px;
          position: relative;
          color: $gui-kit-gray-90;
          text-align: center;
          caption-side: initial;

          &::after {
            content: '';
            position: absolute;
            bottom: 0;
            left: -5px;
            width: calc(100% + 10px);
            height: 1px;
            background-color: #d4d4d4;
          }
        }
        /* stylelint-disable */
        .DayPicker_weekHeader {
          .DayPicker_weekHeader_ul {
            .DayPicker_weekHeader_li {
              small {
                color: $gui-kit-gray-90;
                font-weight: 500 !important;
                font-size: 16px;
              }
            }
          }
        }

        .DayPicker_focusRegion {
          .DayPicker_transitionContainer {
            position: relative;
            overflow: hidden;
            border-radius: 3px;

            &.DayPicker_transitionContainer__horizontal {
              transition: none !important;
            }

            .CalendarMonthGrid {
              background: #fff;
              text-align: left;
              z-index: 0;
              position: absolute;
              left: 9px;

              .CalendarMonthGrid_month__horizontal {
                display: inline-block;
                vertical-align: top;
                min-height: 100%;

                &.CalendarMonthGrid_month__hidden {
                  visibility: hidden;
                }

                &.CalendarMonthGrid_month__hideForAnimation {
                  position: absolute;
                  z-index: -1;
                  opacity: 0;
                  pointer-events: none;
                }

                .CalendarMonth {
                  background: #fff;
                  text-align: center;
                  vertical-align: top;
                  user-select: none;

                  table.CalendarMonth_table {
                    margin-bottom: 0;

                    tbody {
                      tr {
                        td {
                          border: none;
                          border-radius: 50%;
                          font-size: 16px;

                          &:focus {
                            outline: none;
                          }

                          div {
                            width: 35px;
                            height: 35px;
                            display: flex;
                            align-items: center;
                            justify-content: center;
                            margin: auto !important;
                            border-radius: 50%;
                          }

                          &.CalendarDay__default {
                            color: #2a2a2a;
                            padding: 0;
                          }

                          &.CalendarDay__blocked_out_of_range {
                            color: #aaa;
                          }

                          &.CalendarDay__selected {
                            background: transparent !important;
                            color: #fff;

                            div {
                              background: $gui-kit-level-2;
                            }
                          }

                          &.CalendarDay__outside {
                            color: $gui-kit-gray-30;
                          }

                          &:not(.CalendarDay__outside):not(.CalendarDay__blocked_out_of_range):not(.CalendarDay__selected):hover {
                            background: transparent !important;
                            cursor: pointer;

                            div {
                              background: $tc-gray-05;
                            }
                          }
                        }
                      }
                    }
                  }
                }
              }
            }
          }
        }
        /* stylelint-enable */
      }
    }
  }

  &.haveError {
    :global {
      .SingleDatePickerInput {
        .DateInput {
          input {
            border: 2px solid $gui-kit-level-5;
          }
        }
      }
    }
  }
}

.errorMessage {
  @include errorMessage;
}
