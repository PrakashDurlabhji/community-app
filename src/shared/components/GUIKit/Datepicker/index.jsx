/* eslint-disable jsx-a11y/label-has-for */
/**
 * Datepicker component.
 */
import React, { useState } from 'react';
import PT from 'prop-types';
import moment from 'moment';
import './style.scss';

import 'react-dates/initialize';
import { SingleDatePicker } from 'react-dates';
import IconCalendar from 'assets/images/tc-edu/icon-calendar.svg';
import useWindowSize from 'utils/useWindowSize';
import CalendarWeek from 'react-dates/lib/components/CalendarWeek';
import IconNext from '../Assets/Images/icon-next.svg';
import IconPrev from '../Assets/Images/icon-prev.svg';

// eslint-disable-next-line no-unused-expressions, react/forbid-foreign-prop-types
CalendarWeek && CalendarWeek.propTypes && delete CalendarWeek.propTypes.children; // fixing the bug in react-dates, more detail in here https://github.com/airbnb/react-dates/issues/1121

function Datepicker({
  value,
  placeholder,
  label,
  onChange,
  errorMsg,
  required,
}) {
  const [date, setDate] = useState(value ? moment(value) : null);
  const [focused, setFocused] = useState(false);
  const { width } = useWindowSize();
  return (
    <div
      styleName={`container ${date ? 'haveValue' : ''} ${
        errorMsg ? 'haveError' : ''
      } ${focused ? 'isFocused' : ''}`}
    >
      <SingleDatePicker
        hideKeyboardShortcutsPanel
        customInputIcon={<IconCalendar />}
        date={date}
        onDateChange={(changedDate) => {
          setDate(changedDate);
          onChange(changedDate ? changedDate.toDate() : null);
        }}
        focused={focused}
        onFocusChange={({ focused: changedFocused }) => setFocused(changedFocused)
        }
        id={`${value}-${placeholder}-${label}-${errorMsg}-${required}`}
        placeholder={`${placeholder}${placeholder && required ? ' *' : ''}`}
        inputIconPosition="after"
        numberOfMonths={1}
        navPrev={<IconPrev />}
        navNext={<IconNext />}
        displayFormat="MMM DD, YYYY"
        daySize={width > 600 ? 47 : 35}
        renderDayContents={d => (<div>{d.date ? d.date() : ''}</div>)}
        enableOutsideDays
        firstDayOfWeek={1}
        weekDayFormat="ddd"
      />
      {label ? (
        <span styleName="label">
          {label}
          {required ? <span>&nbsp;*</span> : null}
        </span>
      ) : null}
      {errorMsg ? <span styleName="errorMessage">{errorMsg}</span> : null}
    </div>
  );
}

Datepicker.defaultProps = {
  value: null,
  placeholder: '',
  label: '',
  onChange: () => {},
  errorMsg: '',
  required: false,
};

Datepicker.propTypes = {
  value: PT.instanceOf(Date),
  placeholder: PT.string,
  label: PT.string,
  onChange: PT.func,
  errorMsg: PT.string,
  required: PT.bool,
};

export default Datepicker;
