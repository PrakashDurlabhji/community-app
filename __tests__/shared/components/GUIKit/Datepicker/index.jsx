import React from 'react';
import Renderer from 'react-test-renderer/shallow';

import Datepicker from 'components/GUIKit/Datepicker';


const rnd = new Renderer();

it('Default render', () => {
  rnd.render((<Datepicker />));
  expect(rnd.getRenderOutput()).toMatchSnapshot();
});
