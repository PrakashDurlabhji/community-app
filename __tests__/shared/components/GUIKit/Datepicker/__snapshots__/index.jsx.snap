// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default render 1`] = `
<div
  className="src-shared-components-GUIKit-Datepicker-___style__container___Sl58X"
>
  <withStyles(SingleDatePicker)
    anchorDirection="left"
    appendToBody={false}
    block={false}
    calendarInfoPosition="bottom"
    customCloseIcon={null}
    customInputIcon={
      <IconCalendar
        height="16"
        viewBox="0 0 16 16"
        width="16"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    date={null}
    daySize={47}
    disableScroll={false}
    disabled={false}
    displayFormat="MMM DD, YYYY"
    enableOutsideDays={true}
    firstDayOfWeek={1}
    focused={false}
    hideKeyboardShortcutsPanel={true}
    horizontalMargin={0}
    horizontalMonthPadding={13}
    id="null----false"
    initialVisibleMonth={null}
    inputIconPosition="after"
    isDayBlocked={[Function]}
    isDayHighlighted={[Function]}
    isOutsideRange={[Function]}
    isRTL={false}
    keepFocusOnInput={false}
    keepOpenOnDateSelect={false}
    monthFormat="MMMM YYYY"
    navNext={
      <IconNext
        height="16"
        viewBox="0 0 10 16"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    navPrev={
      <IconPrev
        height="16"
        viewBox="0 0 10 16"
        width="10"
        xmlns="http://www.w3.org/2000/svg"
      />
    }
    noBorder={false}
    numberOfMonths={1}
    onClose={[Function]}
    onDateChange={[Function]}
    onFocusChange={[Function]}
    onNextMonthClick={[Function]}
    onPrevMonthClick={[Function]}
    openDirection="down"
    orientation="horizontal"
    phrases={
      Object {
        "calendarLabel": "Calendar",
        "chooseAvailableDate": [Function],
        "clearDate": "Clear Date",
        "closeDatePicker": "Close",
        "dateIsSelected": [Function],
        "dateIsUnavailable": [Function],
        "enterKey": "Enter key",
        "escape": "Escape key",
        "hideKeyboardShortcutsPanel": "Close the shortcuts panel.",
        "homeEnd": "Home and end keys",
        "jumpToNextMonth": "Move forward to switch to the next month.",
        "jumpToPrevMonth": "Move backward to switch to the previous month.",
        "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
        "keyboardShortcuts": "Keyboard Shortcuts",
        "leftArrowRightArrow": "Right and left arrow keys",
        "moveFocusByOneDay": "Move backward (left) and forward (right) by one day.",
        "moveFocusByOneMonth": "Switch months.",
        "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week.",
        "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week.",
        "openThisPanel": "Open this panel.",
        "pageUpPageDown": "page up and page down keys",
        "questionMark": "Question mark",
        "returnFocusToInput": "Return to the date input field.",
        "selectFocusedDate": "Select the date in focus.",
        "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel.",
        "upArrowDownArrow": "up and down arrow keys",
      }
    }
    placeholder=""
    readOnly={false}
    regular={false}
    renderCalendarInfo={null}
    renderDayContents={[Function]}
    renderMonthElement={null}
    renderMonthText={null}
    reopenPickerOnClearDate={false}
    required={false}
    screenReaderInputMessage=""
    showClearDate={false}
    showDefaultInputIcon={false}
    small={false}
    verticalHeight={null}
    verticalSpacing={22}
    weekDayFormat="ddd"
    withFullScreenPortal={false}
    withPortal={false}
  />
</div>
`;
